test/unit/logic/delegate.js

```diff
@@ -0,0 +1,535 @@
+
+	var transactionMock;
+	var accountsMock;
+	var delegate;
+
+	var trs;
+	var rawTrs;
+	var sender;
+
+	function dummyCb () {};
+
+	before(function () {
+		transactionMock = sinon.mock({});
+	});
+
+	beforeEach(function () {
+		trs = _.cloneDeep(validTransaction);
+		rawTrs = _.cloneDeep(rawValidTransaction);
+		sender = _.cloneDeep(validSender);
+
+		accountsMock = {
+			setAccountAndGet: sinon.mock(),
+			getAccount: sinon.mock()
+		};
+
+		delegate = new Delegate(modulesLoader.scope.schema);
+		delegate.bind(accountsMock);
+	});
+
+	//TODO : branch 456-unit-test-for-delegate-transaction will introduce these skipped tests
+	describe.skip('constructor', function () {
+
+		it('should attach schema to library variable', function () {
+			new Delegate(modulesLoader.scope.schema);
+			var library = Delegate.__get__('library');
+
+			expect(library).to.eql({
+				schema: modulesLoader.scope.schema
+			});
+		});
+	});
+
+	describe.skip('bind', function () {
+
+		it('should attach empty object to private modules.accounts variable', function () {
+			delegate.bind({});
+			var modules = Delegate.__get__('modules');
+
+			expect(modules).to.eql({
+				accounts: {}
+			});
+		});
+
+		it('should be okay with correct params', function () {
+			delegate.bind(accountsMock);
+			var modules = Delegate.__get__('modules');
+
+			expect(modules).to.eql({
+				accounts: accountsMock
+			});
+		});
+	});
+
+	describe.skip('calculateFee', function () {
+
+		it('should return the correct fee for delegate transaction', function () {
+			expect(delegate.calculateFee(trs)).to.equal(node.constants.fees.delegate);
+		});
+	});
+
+	describe('verify', function () {
+
+		it('should call callback with error if recipientId exist', function (done) {
+			trs.recipientId = '123456';
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Invalid recipient');
+				done();
+			});
+		});
+
+		it('should call callback with error if amount is not equal to 0', function (done) {
+			trs.amount = 1;
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Invalid transaction amount');
+				done();
+			});
+		});
+
+		it('should call callback with error if sender is already a delegate', function (done) {
+			sender.isDelegate = 1;
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Account is already a delegate');
+				done();
+			});
+		});
+
+		it('should call callback with error if asset is undefined', function (done) {
+			trs.asset = undefined;
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Invalid transaction asset');
+				done();
+			});
+		});
+
+		it('should call callback with error if asset is empty', function (done) {
+			trs.asset = {};
+
+			delegate.verify(trs, sender, function (err) {
+				expect(err).to.equal('Invalid transaction asset');
+				done();
+			});
+		});
+
+		it('should call callback with error if username is does not exist', function (done) {
```

Comment:
and what happens when username is valid? - we should test if the getAccount is called with a proper params then